src/measure_theory/interval_integral.lean

```diff
@@ -1345,4 +1345,128 @@ theorem integral_deriv_eq_sub (hderiv : ∀ x ∈ interval a b, differentiable_a
 integral_eq_sub_of_has_deriv_at (λ x hx, (hderiv x hx).has_deriv_at) hcont'
   (measurable_deriv f).ae_measurable
 
+/-!
+### Integration by parts
+ -/
+
+lemma self_mem_ae_restrict {s : set ℝ} (hs : is_measurable s):
+  s ∈ (measure.restrict measure_space.volume s).ae :=
+begin
+  rw ae_restrict_eq hs,
+  simp only [exists_prop, filter.mem_principal_sets, filter.mem_inf_sets],
+  exact ⟨univ, filter.univ_mem_sets, s, by simp⟩,
+end
+
+lemma integral_congr_Ioc {f g : ℝ → ℝ} (h : ∀ x ∈ Ioc a b, f x = g x) :
+  ∫ x in Ioc a b, f x = ∫ x in Ioc a b, g x :=
+integral_congr_ae (filter.eventually_of_mem (self_mem_ae_restrict is_measurable_Ioc) h)
+
+/--
+If f and g coincide in the relevant integral, their integrals also coincide.
+-/
+lemma interval_integral_congr {f g : ℝ → ℝ} (h : ∀ x ∈ interval a b, f x = g x) :
+ ∫ x in a..b, f x = ∫ x in a..b, g x :=
```

Comment:
Good lemma!  Can you move it up to the section with fundamental facts about the `interval_integral` -- maybe somewhere in lines 300-400ish?  In that section you will be in the namespace `interval_integral`, so can shorten the lemma name to say `integral_congr`.